### PR TITLE
Issue170: Global settings

### LINKED ISSUE #141: Point system
Add max and min numbers of points when creating and editing question. Also have numbers of points recieved upon answering a question depend on how many students have answered the question correctly (based on the equation Larry has in the design document).

### LINKED ISSUE #170: General settings page for admins
Please comment with any settings you wish to be included in this page

### PR DESCRIPTION
**Pull request by esteeroramy**
fixes #170 
fixes #141 
globalsettings

This PR addresses global settings, and the point system
This PR also includes multiple bug fixes.

### CODE REVIEW

File: public/js/settings.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+$(function() {
+    getQuestionsTopicsList();
+});
+
```
Comment (Mohamed-A-Mohamed) [inline public/js/settings.js:4]: Extra line

File: server/questions.js [hunk 2]
```diff
@@ -18,19 +18,25 @@ You should have received a copy of the GNU General Public License
...
 var questionUpdateParser = function(question){
     var updatedQuestion = question;
+
     if ('visible' in question){
         updatedQuestion.visible = (question.visible === 'true');
     }
-    if ('points' in question){
-        updatedQuestion.points = parseInt(question.points);
+
+    if ('minpoints' in question){
```
Comment (Mohammed-JS-Khan) [inline server/questions.js]: space b4 curly

File: server/questions.js [hunk 3]
```diff
@@ -18,19 +18,25 @@ You should have received a copy of the GNU General Public License
...
+        updatedQuestion.minpoints = parseInt(question.minpoints);
+    }
+
+    if ('maxpoints' in question){
```
Comment (Mohammed-JS-Khan) [inline server/questions.js]: space b4 curly

File: server/questions.js [hunk 4]
```diff
@@ -105,7 +112,7 @@ var prepareQuestionData = function(question, callback){
 /*
 * Insert a new regular question into the database.
 * The question object passed to the function should have
-* the text, topic, type, answer, points and hint set.
+* the text, topic, type, answer, minpoints, maxpoints and hint set.
 */
 exports.addQuestion = function(question, callback) {
     prepareQuestionData(question, function(err, questionToAdd){
```
Comment (Mohammed-JS-Khan) [inline server/questions.js]: space b4 curly

**Review by Mohammed-JS-Khan — APPROVED** [review]

File: scripts/dataGenerator.js [hunk 5]
```diff
@@ -343,12 +352,15 @@ var createQuestionsTrueFalse = function () {
  * @param {integer} id
  */
 var addQuestionTrueFalse = function (qTopic, id) {
```
Comment (Mohamed-A-Mohamed) [inline scripts/dataGenerator.js:354]: I know you didn't make the functions for the questions (ex addQuestionTrueFalse) but do you think it'd be possible to have just one question function and then have another function to change the type of each question so we have less duplicated code?

@esteeroramy @mehtadarshan @ahmedqarmout @larryyueli @Mohammed-JS-Khan

File: server/db.js [hunk 6]
```diff
@@ -18,22 +18,24 @@ You should have received a copy of the GNU General Public License
...
 
-var db = new Db(DB_NAME, new Server(DB_HOST, DB_PORT));
+const db = new Db(DB_NAME, new Server(DB_HOST, DB_PORT));
 
-var nextQuestionNumber = 0;
 var usersCollection;
 var questionsCollection;
 var analyticsCollection;
+var settingsCollection;
+
```
Comment (Mohamed-A-Mohamed) [inline server/db.js:37]: Extra line

**Review by Mohamed-A-Mohamed — APPROVED** [review]
Very nice work :+1:

File: scripts/dataGenerator.js (hunk 5 above)
Comment (esteeroramy) [reply scripts/dataGenerator.js:354]: Well, this is the Datagenerator so efficiency here is not our main concern. The issue is that the processes are async, and so you can only add a type of question, answer it, then add the next one, etc... And so with this design each function waits on the one before it in order to run

File: public/js/settings.js (hunk 1 above)
Comment (esteeroramy) [reply public/js/settings.js:4]: Fixed

File: server/questions.js (hunk 2 above)
Comment (esteeroramy) [reply server/questions.js]: Fixed

File: server/questions.js (hunk 3 above)
Comment (esteeroramy) [reply server/questions.js]: Fixed

File: server/questions.js (hunk 4 above)
Comment (esteeroramy) [reply server/questions.js]: Fixed

File: server/db.js (hunk 6 above)
Comment (esteeroramy) [reply server/db.js:37]: This extra line is to separate the variables

**Review by mehtadarshan1 — APPROVED** [review]

**Pull request merged**